### ISSUE TITLE
Support environments outside of K8s for the tracing endpoint returned by tempo-k8s's tracing charm lib

### ISSUE THREAD

**Issue by shayancanonical** [canonical/tempo-k8s-operator#110]
### Bug Description

When we call the `self.tracing.oltp_http_endpoint()` method (where self.tracing is an instance of TracingEndpointRequirer), we get an endpoint (`tempo-0.tempo-endpoints.cos.svc.cluster.local`) that is only resolvable in the K8s cluster. If I have a cross-cloud relation with mysql in an lxd cloud, then the returned endpoint is not resolvable.

I also suspect that something similar may happen with a cross cloud relation when the client application is deployed in a separate K8s cluster (because of the `.cluster.local` in the returned endpoint)

### To Reproduce

1. Deploy COS in a k8s model
2. Attempt to ping the tempo unit `tempo-0.tempo-endpoins.<cos-model-name>.svc.cluster.local` from outside of the K8s cluster

### Environment

microk8s: v1.27.13
tempo-k8s revision: 30
tracing charm-libs: tempo-k8s/v1/tracing.py and tempo-k8s/v1/charm_tracing.py

### Relevant log output

```shell
shayanp@brisbane:~$ ping tempo-0.tempo-endpoints.cos.svc.cluster.local
ping: tempo-0.tempo-endpoints.cos.svc.cluster.local: Temporary failure in name resolution
```

**Comment by mmkay** [comment]
Hi @shayancanonical! Latest revision with v2 tracing will already fix it for you (see #94 for reference) however I'd suggest to wait with the upgrade till #108 is merged as we're doing final touches to the protocol.

**Comment by mmkay** [comment]
Should now be fixed with the latest `tracing v2` and latest tempo revision. Please reopen if it still appears on your environment.
